sources/org.osbuild.curl

```diff
@@ -86,6 +87,59 @@ SCHEMA = """
 """
 
 
+class Subscriptions:
+    def __init__(self, sections):
+        self.sections = sections
+
+    @classmethod
+    def from_host_system(cls):
+        with open('/etc/yum.repos.d/redhat.repo', 'r') as fp:
+            return cls._parse_repo_file(fp)
+
+    @staticmethod
+    def _process_baseurl(input_url):
+        # First escape meta characters: . ^ $ * + ? { } [ ] \ | ( )
+        # Except for $basearch, which will be replaced later.
+        for character in ['\\', '.', '^', '*', '+', '?', '{', '}', '[', ']', '|', '(',')']:
+            input_url = input_url.replace(character, f"\\{character}")
+
+        # Now replace variables with regexes (see man 5 yum.conf for the list)
+        for variable in ['$releasever', '$arch', '$basearch', '$uuid']:
+            input_url = input_url.replace(variable, '.*')
+
+        if input_url.endswith('/'):
+            input_url = input_url + '.*'
+        else:
+            input_url = input_url + '/.*'
+
+        return re.compile(input_url)
+
+    @classmethod
+    def _parse_repo_file(cls, fp):
+        parser = configparser.ConfigParser()
+        parser.read_file(fp)
+
+        sections = dict()
+        for section in parser.sections():
+            sections[section]=dict()
+            sections[section]['baseurl']=cls._process_baseurl(parser.get(section, 'baseurl'))
+            for parameter in ['sslcacert', 'sslclientkey', 'sslclientcert']:
+                sections[section][parameter]=parser.get(section, parameter)
+
+        return cls(sections)
+
+    def get_secrets(self, rpm_url):
+        for _, parameters in self.sections.items():
+            if parameters['baseurl'].match(rpm_url) is not None:
+                return {
+                    'ssl_ca_cert': parameters['sslcacert'],
+                    'ssl_client_key': parameters['sslclientkey'],
+                    'ssl_client_cert': parameters['sslclientcert']
+                }
+
+        raise RuntimeError(f'There are no RHSM secret associated with {rpm_url}')
```

Comment:
A cosmetic suggestion. Although the use of `"` and `'` is a bit mixed in the code, I would say that it would be more consistent to use `"` here.

Reply:
I created a brand new module for the class, so it is not mixed any more :-)

Reply:
Fair enough :laughing: